artifact.md

```diff
@@ -1,5 +1,7 @@
 # OCI Artifact Manifest Specification
 
+Implementations concerned with portability SHOULD NOT use the Artifact Manifest.
```

Comment:
"portability" would need to be explicitly defined for discussing this PR further

Reply:
Good point. I have in mind "portability" between 1.0 and 1.1 registries, so perhaps "backwards compatibility with OCI 1.0" is a better phrase? We could link to a section of image manifest describing its use for packaging non-container-image content.

Reply:
I agree with that messaging change. It should be OK for an implementation to decide its own destiny if they've already adopted OCI 1.0 image spec and they want to phase it out at a major version change, their compatibility story _is their own business_.

The same applies for all future versions, and in fact, **all** future image-spec types!

How about this in spec.md or in all manifest types?

> Implementations that support creating alternative manifest types SHOULD NOT use the registry to determine the manifest type created.

@sudo-bmitch does that address your objection about non-deterministic behavior? I should be able to look at a tool like `notary@vX.Y` and its command line arguments and _know_ what it will produce. Tools that are clever and try to automatically fall back will end up surprising users.

And understanding @sudo-bmitch's concern, I'll also close this issue:
- https://github.com/opencontainers/distribution-spec/issues/368

I don't think it's a good idea to specify fallback behavior any longer.

Reply:
@sudo-bmitch re:

> I'm not sure that a warning is enough if OCI is tasked with portability and compatibility. I'm leaning towards saying that any tooling that produces the artifact manifest, without an experimental flag or some other end-user control

Such as a user intentionally upgrading to a new major version, i.e.: OCI client Foobarctl says "with major version v3, we default to artifact now and will fail on pushing to an OCI 1.0 registry. Opt out of this behavior by passing `--oci-spec=v1.0`".

Reply:
After the conversations yesterday, some kind of verbiage could be useful. Let's steer to something hand-wavy, but maybe
```
**Introduced**: v1.1.0

_Notice_: Introduction of new manifest types requires client and server features. As implementations adopt a new manifest type it may cause portability challenges with legacy tools or services.
```

Reply:
I like the wording that @vbatts provides, with the addition that I'd like to include a "SHOULD NOT use" suggestion in there to stress our desire for portability.